cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1155,14 +1155,48 @@
 
 ;;;; Rules for `iabs` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(rule 1 (lower (has_type $I8X16 (iabs x)))
```

Comment:
Can you remind me what `1` means here? Does it mean "higher priority", so the SSSE3 condition is checked _before_ we emit the long sequence?

Reply:
Indeed! Rules are match in-order highest-to-lowest priority and the first matching one is the one used for lowering. Rules then aren't allowed to overlap at all which is why a priority is required here because otherwise if `use_ssse3` is `true` then it's ambiguous which rule to use here (so the priority says check this one first)